fetchDischarge.R

```diff
@@ -28,37 +30,38 @@ fetch.discharge <- function(viz){
       currentSites <- readNWISstat(siteNumbers = getSites,
                                    parameterCd = "00060", 
                                    statReportType="daily",
-                                   statType=c("p10","p25","p50","p75","p90","mean"),
-                                   startDate = startDate, endDate = endDate)
+                                   statType=c("p10","p25","p50","p75","p90","mean"))
       statData <- rbind(statData,currentSites)
     }
     
-    finalJoin <- left_join(storm.data,statData)
+    #NOTE: won't deal with crossing months
+    statData.storm <- statData[statData$month_nu == month(startDate) & 
+                                 statData$day_nu >= day(startDate) & 
+                                 statData$day_nu <= day(endDate),]
+                                 statData$day_nu <= day(endDate),]
+    
+    finalJoin <- left_join(storm.data,statData.storm)
     finalJoin <- left_join(finalJoin,sites) 
     
     #remove sites without current data 
-    finalJoin <- finalJoin[!is.na(finalJoin$Flow),] 
-    
+    #finalJoin <- finalJoin[!is.na(finalJoin$Flow),] 
+    return(finalJoin)
   }#end
+  
   library(dataRetrieval)
   library(dplyr)
-  #TODO: call it
+  library(lubridate)
   
-}
-
-
-
-
-#retrieve stats data, dealing with 10 site limit to stat service requests
+  #TODO: timezones
+  startDate <- as.Date("2016-10-5")
+  endDate <- as.Date("2016-10-6")
+  states <- c("FL","GA","SC","NC")
```

Comment:
we need to figure out how to not declare the states in a bunch of different places.